### PR TITLE
yoshi-typescript: add some tests and fix bug

### PR DESCRIPTION
**Pull request by ranyitz**
the bug was with typescript passing when it actually failing.
https://jira.wixpress.com/browse/CI-8803

### CODE REVIEW

File: plugins/yoshi-typescript/test/index.spec.js [hunk 1]
```diff
@@ -59,23 +62,62 @@ describe('Typescript', () => {
...
 
-  it('should fail with exit code 1', () => {
+  it('should resolve when exist code is 0', () => {
+    test.setup({
+      'src/a.ts': 'const a = 1',
+      'tsconfig.json': tsconfig()
+    });
+
+    return expect(typescript({watch: false, log: identity})()
+      ).to.be.fulfilled;
```
Comment (ronami) [inline plugins/yoshi-typescript/test/index.spec.js]: weird indentation

File: plugins/yoshi-typescript/test/index.spec.js [hunk 2]
```diff
@@ -59,23 +62,62 @@ describe('Typescript', () => {
...
+  it('should reject when there is a typescript error and not in watch mode', () => {
+    test.setup({
+      'src/a.ts': 'function ()',
+      'tsconfig.json': tsconfig()
+    });
+
+    return typescript({watch: false, log: identity})()
       .catch(() => {
-        expect(stdout).to.contain('error TS1003: Identifier expected');
+        expect(stdout).to.contain('error TS');
```
Comment (ronami) [inline plugins/yoshi-typescript/test/index.spec.js]: test will pass if `typescript(...)()` returns a resolved promise

File: plugins/yoshi-typescript/test/index.spec.js [hunk 3]
```diff
@@ -59,23 +62,62 @@ describe('Typescript', () => {
...
         expect(test.content('dist/app/a.js')).to.contain('//# sourceMappingURL=a.js.map');
         expect(test.list('dist/app')).to.include('a.js.map', 'a.d.ts');
       });
   });
 
-  it('should fail with exit code 1', () => {
+  it('should resolve when exist code is 0', () => {
+    test.setup({
+      'src/a.ts': 'const a = 1',
+      'tsconfig.json': tsconfig()
```
Comment (ronami) [inline plugins/yoshi-typescript/test/index.spec.js]: we only use `tsconfig()` without passing any params, it doesn't need to be a function.

File: plugins/yoshi-typescript/test/index.spec.js [hunk 4]
```diff
@@ -59,23 +62,62 @@ describe('Typescript', () => {
...
+      ).to.be.fulfilled;
+  });
+
+  it('should resolve when it is in watch mode and there are no errors', () => {
+    test.setup({
+      'src/a.ts': 'const a = 1',
+      'tsconfig.json': tsconfig()
+    });
+
+    return expect(typescript({watch: true, log: identity})()
```
Comment (ronami) [inline plugins/yoshi-typescript/test/index.spec.js]: I would consider splitting the `typescript(...)()` expression into more than one line to make it more readable

**Comment by ranyitz** [comment]
Fixed, waiting for approve @ronami

**Comment by ronami** [comment]
lgtm, merging

**Pull request merged**